Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -470,40 +478,94 @@ void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
   // The q-vector direction of the peak is = goniometer * ub * hkl_vector
   // This is in inelastic convention: momentum transfer of the LATTICE!
   // Also, q does have a 2pi factor = it is equal to 2pi/wavelength.
-  V3D q = orientedUB * hkl * (2.0 * M_PI * m_qConventionFactor);
+  const auto q = orientedUB * hkl * (2.0 * M_PI * m_qConventionFactor);
+  const auto params = getPeakParametersFromQ(q);
+  const auto detectorDir = std::get<0>(params);
+  const auto wl = std::get<1>(params);
+
+  const bool useExtendedDetectorSpace =
+      getProperty("PredictPeaksOutsideDetectors");
+  const auto result = m_detectorCacheSearch->findDetectorIndex(q);
+  const auto hitDetector = std::get<0>(result);
+  const auto index = std::get<1>(result);
 
-  // Create the peak using the Q in the lab framewith all its info:
-  Peak p(m_inst, q);
-  if (m_edge > 0) {
-    if (edgePixel(m_inst, p.getBankName(), p.getCol(), p.getRow(), m_edge))
+  if (!hitDetector && !useExtendedDetectorSpace) {
+    return;
+  }
+
+  const auto &detInfo = m_pw->detectorInfo();
+  const auto &det = detInfo.detector(index);
+  std::unique_ptr<Peak> peak;
+
+  if (hitDetector) {
+    // peak hit a detector to add it to the list
+    peak = Kernel::make_unique<Peak>(m_inst, det.getID(), wl);
+    if (!peak->getDetector())
       return;
+
+  } else if (useExtendedDetectorSpace) {
+    // use extended detector space to try and guess peak position
+    const auto component =
+        m_inst->getComponentByName("extended-detector-space");
+    const auto c = boost::dynamic_pointer_cast<const ObjComponent>(component);
```

Comment:
What is c? I would also avoid single character names as they are prone to typos.